### PR TITLE
Fixed #26927 -- Fixed subwidget iteration to pass disabled and required attributes.

### PR DESCRIPTION
**Pull request by jdufresne**
<https://code.djangoproject.com/ticket/26927>

### CODE REVIEW

File: django/forms/widgets.py [hunk 1]
```diff
@@ -787,6 +787,19 @@ class CheckboxSelectMultiple(RendererMixin, SelectMultiple):
...
+        attrs = super(CheckboxSelectMultiple, self).build_attrs(extra_attrs, **kwargs)
+        # The required attribute comes from the bound field if the form field
+        # is required. When required is set on multiple checkboxes with the
+        # same name, the browser will require each checkbox be checked to
+        # submit the form. This is not the desired validation, so remove the
+        # attribute.
+        try:
+            del attrs['required']
+        except KeyError:
+            pass
```
Comment (akki) [inline django/forms/widgets.py]:
```python
if 'required' in attrs:
  del attrs['required']
```
or
`attrs.pop('required', None)` might be better here unless I am missing something and you want to handle `KeyError` specifically.

Comment (jdufresne) [inline django/forms/widgets.py]: In these cases, I tend to follow the suggested Python style [easier to ask for forgiveness than permission](https://docs.python.org/3/glossary.html#term-eafp). But if the Django project prefers a different style, I have no problem changing it.

Given this, would you still prefer I change this?

Comment (charettes) [inline django/forms/widgets.py]: I think the motto apply in the case where you let the exception propagate, when you catch it anyway I see no harm in using `attrs.pop('required', None)` instead.

Comment (akki) [inline django/forms/widgets.py]: Ahh! No strong opinion in that case. Thanks!

**Comment by jdufresne** [comment]
Made suggested change. All additional feedback welcome. Thanks.

File: django/forms/boundfield.py [hunk 2]
```diff
@@ -227,3 +225,14 @@ def id_for_label(self):
         widget = self.field.widget
         id_ = widget.attrs.get('id') or self.auto_id
         return widget.id_for_label(id_)
+
+    def build_widget_attrs(self, attrs, widget=None):
+        if not widget:
+            widget = self.field.widget
+
+        attrs = dict(attrs)
```
Comment (timgraham) [inline django/forms/boundfield.py]: Needed? No tests are failing with it removed.

**Comment by timgraham** [comment]
Doc edits: http://dpaste.com/1GG82GK

File: django/forms/boundfield.py (hunk 2 above)
Comment (timgraham) [inline django/forms/boundfield.py]: I added a comment that this is needed to avoid modifying the argument value.

**Comment by timgraham** [comment]
merged in ac3aaaa740dcf9c6efd2f88ee9219c1924c7695e, thanks!

**Pull request closed without merging**